lib/modules/ai_bot/commands/search_command.rb

```diff
@@ -120,12 +124,35 @@ def process(**search_args)
         )
 
       # let's be frugal with tokens, 50 results is too much and stuff gets cut off
-      limit ||= 20
-      limit = 20 if limit > 20
+      limit ||= MAX_RESULTS
+      limit = MAX_RESULTS if limit > MAX_RESULTS
+
+      should_try_semantic_search = SiteSetting.ai_embeddings_semantic_search_enabled
+      should_try_semantic_search &&= (limit == MAX_RESULTS)
+      should_try_semantic_search &&= (search_args.keys - %i[search_query order]).length == 0
+      should_try_semantic_search &&= (search_args[:search_query].present?)
+
+      limit = limit - MIN_SEMANTIC_RESULTS if should_try_semantic_search
 
       posts = results&.posts || []
       posts = posts[0..limit - 1]
 
+      if should_try_semantic_search
+        semantic_search = DiscourseAi::Embeddings::SemanticSearch.new(Guardian.new())
+        topic_ids = Set.new(posts.map(&:topic_id))
```

Comment:
This is a nice way of getting uniq, I am guessing it's faster than doing `.uniq` on the end?

Reply:
It is really splitting hairs cause there are just 20 elements here... but I wanted a Set ... not an Array so the `include?` is faster.